x-pack/plugins/monitoring/server/routes/api/v1/elasticsearch_settings/check/internal_monitoring.ts

```diff
@@ -4,46 +4,97 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { CatIndicesParams } from 'elasticsearch';
+// @ts-ignore
+import { RequestHandlerContext } from 'kibana/server';
+import { getIndexPatterns } from '../../../../../lib/cluster/get_index_patterns';
+
 // @ts-ignore
 import { handleError } from '../../../../../lib/errors';
 import { RouteDependencies } from '../../../../../types';
 
-export function internalMonitoringCheckRoute(_server: unknown, npRoute: RouteDependencies) {
+const queryBody = {
+  size: 15,
+  query: {
+    bool: {
+      filter: [
+        {
+          range: {
+            timestamp: {
+              gte: 'now-15m',
+              lte: 'now',
+            },
+          },
+        },
+      ],
+    },
+  },
+  sort: [
+    {
+      timestamp: {
+        order: 'desc',
+      },
+    },
+  ],
+  _source: {
+    excludes: ['*'],
+  },
+};
+
+const checkLatestMonitoringIsLegacy = async (context: RequestHandlerContext, index: string) => {
+  const { client: esClient } = context.core.elasticsearch.legacy;
+  const result = await esClient.callAsCurrentUser('search', {
+    index,
+    body: queryBody,
+  });
+
+  const {
+    hits: { hits },
+  } = result;
+  const counts = {
+    legacyIndicesCount: 0,
+    mbIndicesCount: 0,
+  };
+
+  if (!hits?.length) {
+    return counts;
+  }
+
+  counts.mbIndicesCount = hits.filter(({ _index }: { _index: string }) =>
+    _index.includes('-mb-')
+  ).length;
+
+  counts.legacyIndicesCount = hits.length - counts.mbIndicesCount;
```

Comment:
These counts seem off to me.

I added this debug code:
`console.log(hits.map((h) => h._index));`

which results in:

```
[ '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21',
  '.monitoring-kibana-7-2020.07.21' ]
```

Maybe you can just do a query like:
```
POST .monitoring-es-*/_search
{
  "size": 0,
  "aggs": {
    "types": {
      "terms": {
        "field": "_index",
        "size": 10
      }
    }
  }
}
```